modules/imgproc/src/imgwarp.cpp

```diff
@@ -2713,10 +2984,9 @@ class WarpPerspectiveInvoker :
         int bw0 = std::min(BLOCK_SZ*BLOCK_SZ/bh0, width);
         bh0 = std::min(BLOCK_SZ*BLOCK_SZ/bw0, height);
 
-        #if CV_TRY_SSE4_1
-        Ptr<opt_SSE4_1::WarpPerspectiveLine_SSE4> pwarp_impl_sse4;
-        if(CV_CPU_HAS_SUPPORT_SSE4_1)
```

Comment:
This part is necessary to dynamically call to specific implementation in case SSE4_1-instructions are available.
It looks like existing performance for SSE4 is better than UI-based performance for SSE2 for linear interpolation(at least for some cases) so IMO it make sense to retain dynamic dispatching of opt_SSE4_1::WarpPerspectiveLine_SSE4 here.
@alalek What's your opinion on this?

Reply:
Lets keep SSE4_1 optimization here.

What is about performance difference with AVX2 baseline?

Reply:
AVX2 baseline is better than SSE4_2 baseline both before and after UI implementation. AVX2 baseline also provide better performance improvement. However AVX2 baseline before UI implementation provide worse performance than SSE4_2 after. So IMO it make sense to implement UI-related dynamic dispatching for the module.

Reply:
Yes. Checking for `CV_CPU_HAS_SUPPORT_SSE4_1` allows to decide on SSE4 availability at runtime and dispatch execution accordingly. Preprocessor check for `CV_TRY_SSE4_1` is performed to decide whether dynamic dispatching of the code is possible and necessary(AFAIK this feature is implemented for x86/64 only and could be completely disabled by the user for some reason)

Reply:
I think it would be better to retain SSE4_1 implementation the way it was to make this PR cleaner

Reply:
Re-add dynamic runtime and dispatch execution.